src/components/ui/InfiniteList/index.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useRef, useEffect, useState, HTMLAttributes, ReactNode } from 'react';
+
+import { StyledInfiniteList } from './Styled';
+import { BaseProps } from '../Base';
+import { Cog as Spinner } from '../icons'; // TODO: update icon
+
+export interface InfiniteListProps extends Omit<HTMLAttributes<HTMLUListElement>, 'css'>, BaseProps {
+    /* The elements that will populat the list. These elements will be wrapped in <li> tags, so that can be any element. */
+    items: ReactNode[];
+    /* A callback function that will make an api call to load the next batch of items. */
+    handleLoad: () => void;
+    /* The number of items that will be loaded on each load. */
+    batchSize: number;
+}
+
+const InfiniteList = (props: InfiniteListProps) => {
+    const { items, handleLoad, batchSize } = props;
+
+    const feedRef = useRef<HTMLUListElement>(null);
+    const listEnd = useRef<HTMLDivElement>(null);
+    const currentTopItem = useRef<HTMLLIElement>(null);
+    const firstNew = useRef<HTMLLIElement>(null);
+    const prevLength = useRef(items.length)
+    const newLength = useRef(0);
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    // Scroll to the bottom of the list on load
+    useEffect(() => {
+        feedRef.current?.addEventListener('scroll', handleScroll);
+        scrollToBottom();
+        return feedRef.current?.removeEventListener('scroll', handleScroll, true)
```

Comment:
`return () => feedRef.current.remove...`, right?

Reply:
Dang it, yes. That gets me every time.